src/getInfo.ts

```diff
@@ -1,22 +1,23 @@
-export default function getInfo(data) {
-  const { x, y } = data;
+interface DataType {
+  x: number[];
+  y: number[];
+}
+
+export default function getInfo({ x, y }: DataType) {
   let x0 = { x: x[0], y: y[0] };
   let y0 = { x: x[0], y: y[0] };
   let max = { x: x[0], y: y[0] };
-  let power = { x, y: [] };
+  let power: DataType = { x, y: [] };
   for (let i = 0; i < x.length; i++) {
     if (Math.abs(y[i]) < Math.abs(y0.y)) {
-      y0.x = x[i];
-      y0.y = y[i];
+      y0 = { x: x[i], y: y[i] };
     }
     if (Math.abs(x[i]) < Math.abs(x0.x)) {
-      x0.x = x[i];
-      x0.y = y[i];
+      x0 = { x: x[i], y: y[i] };
     }
     power.y.push(x[i] * y[i]);
     if (x[i] * y[i] < max.x * max.y) {
-      max.x = x[i];
-      max.y = y[i];
+      max = { x: x[i], y: y[i] };
```

Comment:
Why do you want to create a new object ? Seems more difficult for the garbage collector.